internal/operators/ocs/ocs_operator.go

```diff
@@ -92,6 +95,54 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(context.Context, *common.Cluster, *models.Host) (*models.ClusterHostRequirementsDetails, error) {
-	return &models.ClusterHostRequirementsDetails{}, nil
+func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	numOfHosts := len(cluster.Hosts)
+
+	inventory, err := hostutil.UnmarshalInventory(host.Inventory)
+	if err != nil {
+		return nil, err
+	}
+
+	disks := int64(getValidDiskCount(inventory.Disks) - 1)
+
+	if numOfHosts < 3 {
+		return nil, fmt.Errorf("number of hosts is less than 3")
+	}
+
+	if numOfHosts == 3 { // Compact Mode
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
+			if disks < 0 {
+				return nil, fmt.Errorf("in compact mode each host should have at least 1 disk")
+			}
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores:   CPUCompactMode + disks*o.config.OCSRequiredDiskCPUCount,
+				RAMMib:     conversions.GbToMib(MemoryGBCompactMode + disks*o.config.OCSRequiredDiskRAMGB),
+				DiskSizeGb: MinDiskSize,
+			}, nil
+		}
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	}
+
+	if host.Role == models.HostRoleAutoAssign { //If auto-assign we cannot proceed with minimal or standard mode
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	}
+
+	// Minimum mode is the minimum version of the standard mode. If the resources will be more, standard mode will be deployed automatically.
```

Comment:
Please reword comment: The OCS minimal deployment mode sets up a storage cluster using reduced resources.